src/python/upload.py

```diff
@@ -54,8 +54,20 @@ def upload(obj_dict, obj_type, excel_uploader_object, gui_object):
     for key, obj in obj_dict.items():
 
         try:
-            if obj_type == "File" and obj.name is None:
-                obj.name = obj.source
+            if obj_type == "File":
+                # iterate through list of files
+                for file in obj:
+                    if file.name is None:
+                        file.name = file.source
```

Comment:
this part might be problematic because would the file name be `https://my-file-source.com` or `C:\Users\navid\OneDrive\Desktop\test_file_upload` ?

Reply:
This is only used for website links

Reply:
got it!

I think that should work for now, but we should document this for users so they are aware